src/three-components/Model.js

```diff
@@ -114,11 +123,37 @@ export default class Model extends Object3D {
       return;
     }
 
-    const scene = await this.loader.load(url);
+    // If we have pending work due to a previous source change in progress,
+    // cancel it so that we do not incur a race condition:
+    if (this[$cancelPendingSourceChange] != null) {
```

Comment:
Similar concept to other features ensuring that the changes are still valid; can that be abstracted so that model can handle it everywhere? No good ideas currently, but this seems like a potential growing source of race conditions throughout features

Reply:
I disagree, this problem is fairly unique to the consequences of changing the `src`. Let's discuss offline.

Reply:
After reviewing, I remain convinced that changing the `src` requires special considerations. I'm open to concrete suggestions for refactoring, though.